test/classes.coffee

```diff
@@ -1808,3 +1808,19 @@ test 'Bound method of immediately instantiated class with expression base class
 
   {derivedBound} = a
   eq derivedBound(), 3
+
+test "#4591: super.x.y, super['x'].y", ->
```

Comment:
Usually `super.foo` refers to a method named `foo`, as in `super.foo()`. We should add another test that tests `super.x.y()` and `super['x'].y()`